src/Yasumi/Provider/USA.php

```diff
@@ -161,9 +161,18 @@ private function calculateMemorialDay(): void
     private function calculateJuneteenth(): void
     {
         if ($this->year >= 2021) {
+            $date = new DateTime("$this->year-6-19", DateTimeZoneFactory::getDateTimeZone($this->timezone));
+            $label = 'Juneteenth';
+            if ('Sunday' === $date->format('l')) {
+                $date->add(DateInterval::createfromdatestring('+1 day'));
+                $label .= ' (observed)';
+            } elseif ('Saturday' === $date->format('l')) {
+                $date->add(DateInterval::createfromdatestring('-1 day'));
+                $label .= ' (observed)';
+            }
             $this->addHoliday(new Holiday('juneteenth', [
-                'en' => 'Juneteenth',
-            ], new DateTime("$this->year-6-19", DateTimeZoneFactory::getDateTimeZone($this->timezone)), $this->locale));
+            'en' => $label,
+          ], $date, $this->locale));
```

Comment:
For holidays that are observed/substituted due to that they may fall in the weekend for example, you should use the `SubstituteHoliday` class.